src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -563,9 +576,9 @@ void InferRequest::wait() {
             {
```

Comment:
Can we combine `outputMemory != nullptr` check with `if (!bptr->is<gpu::ClBlob>()) {`?

Reply:
Not sure this is what you meant, but I did some more refactoring. https://github.com/openvinotoolkit/openvino/pull/15631/commits/70499d41d5efb9723c59b4c47749db23d0416418

Reply:
My suggestion was to extend [this condition](https://github.com/openvinotoolkit/openvino/blob/70499d41d5efb9723c59b4c47749db23d0416418/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L568) with additional `outputMemory != nullptr` and remove it from [line 573](https://github.com/openvinotoolkit/openvino/blob/70499d41d5efb9723c59b4c47749db23d0416418/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L573) and [line 575](https://github.com/openvinotoolkit/openvino/blob/70499d41d5efb9723c59b4c47749db23d0416418/src/plugins/intel_gpu/src/plugin/infer_request.cpp#L575)

Reply:
Ah, I see. I updated : )